core/src/main/java/org/apache/iceberg/AllDataFilesTable.java

```diff
@@ -42,10 +42,16 @@
 public class AllDataFilesTable extends BaseMetadataTable {
   private final TableOperations ops;
   private final Table table;
+  private final String name;
 
   public AllDataFilesTable(TableOperations ops, Table table) {
```

Comment:
We instantiate some metadata tables in tests so I kept the old constructor too. Won't harm if someone is using it directly too.

Reply:
Does it need to be public? Maybe we should make the constructors package-private while we're making these changes. What do you think?